db/test/test_dataset.py

```diff
@@ -139,6 +139,34 @@ def test_last_edited(self):
         ds_updated = dataset.get(id)
         self.assertTrue(ds_updated['last_edited'] > ds['last_edited'])
 
+    def test_create_snapshot(self):
+        id = dataset.create_from_dict(self.test_data, author_id=self.test_user_id)
+        snapshots = dataset.get_snapshots_for_dataset(id)
+        self.assertEqual(len(snapshots), 0)
+
+        snap_id = dataset.create_snapshot(id)
+        snapshots = dataset.get_snapshots_for_dataset(id)
+        self.assertEqual(len(snapshots), 1)
+        self.assertEqual(snapshots[0]["id"], snap_id)
+
+        snap = dataset.get_snapshot(snap_id)
+        self.assertEqual(snap["data"], self.test_data)
```

Comment:
I've left this `assert` failing, so that we can decide if we want this to be true.
`test_data` has a `"public": True` field which is not in the snapshot. The snapshot has `"id": n` fields in each class. Does it make sense to include/not include these fields in the snapshot, therefore making the dict equal, or are `dataset_class.id` values useful in the snapshot data?